pif_ingestor/core.py

```diff
@@ -1,42 +1,65 @@
 from .ui import get_cli
-from .manager import run_extension
+from .manager import IngesterManager
 from .enrichment import add_tags, add_license, add_contact
 from .uploader import upload
 from .packager import create_package
 import os.path
+from os import walk
 from pypif import pif
 import logging
 
 
-def main():
-    """Main driver for pif-ingestor"""
-
-    # Get direction from the user
-    parser = get_cli()
-    args = parser.parse_args()
-
+def _handle_pif(path, ingest_name, convert_args, enrich_args, ingest_manager):
+    """Ingest and enrich pifs from a path, returning affected paths"""
     # Run an ingest extension
-    pifs = run_extension(args.format, args.path, args.converter_arguments)
+    pifs = ingest_manager.run_extension(ingest_name, path, convert_args)
 
     # Perform enrichment
-    add_tags(pifs, args.tags)
-    add_license(pifs, args.license)
-    add_contact(pifs, args.contact)
+    add_tags(pifs, enrich_args['tags'])
+    add_license(pifs, enrich_args['license'])
+    add_contact(pifs, enrich_args['contact'])
 
     # Write the pif
-    if os.path.isfile(args.path):
-        pif_name = os.path.join(os.path.dirname(args.path), "pif.json")
+    if os.path.isfile(path):
+        pif_name = os.path.join(os.path.dirname(path), "pif.json")
+        res = [path, pif_name]
     else:
-        pif_name = os.path.join(args.path, "pif.json")
+        pif_name = os.path.join(path, "pif.json")
```

Comment:
What is preventing a name conflict if there are two files that are peers in a directory that each generate pifs?